### ISSUE TITLE
Allow refreshing.

### ISSUE THREAD

**Issue by conartist6** [faceyspacey/redux-first-router#345]
There should be a simple API for refreshing the current route's data, which is to say re-triggering its thunk. This is a useful bit of functionality to be able to expose directly to users, e.g. by having a button somewhere in your site's layout that will reload data for whatever page is active. It is also important to be able to do programmatically, for example when some update is known to have caused the page's data to become stale.

**Comment by faceyspacey** [comment]
i like that feature idea.

**Comment by hedgepigdaniel** [comment]
I think it would be fairly simple to do with the existing - assuming you have `redux-thunk` set up:

```javascript
const refreshCurrentRoute = (dispatch, getState) => {
  const currentLocation = {
    type: getState().location.type,
    payload: getState().location.payload,
    // ...etc
    // maybe if you do something to avoid unecessary data loading you also need a flag to ensure that the refresh actually happens
  };
  dispatch(redirect(currentLocation));
};
```

Then if you want to refresh the route data, you can `dispatch(refreshCurrentRoute)`

**Comment by ScriptedAlchemy** [comment]
Agreed, this is an effective way to do it with the existing API, it would be nice to abstract it into the core though
